cpp/src/strings/copying/copying.cu

```diff
@@ -33,47 +34,49 @@ namespace cudf {
 namespace strings {
 namespace detail {
 
-std::unique_ptr<cudf::column> copy_slice(strings_column_view const& strings,
+std::unique_ptr<cudf::column> copy_slice(strings_column_view const& input,
                                          size_type start,
                                          size_type end,
                                          rmm::cuda_stream_view stream,
                                          rmm::mr::device_memory_resource* mr)
 {
-  if (strings.is_empty()) return make_empty_column(type_id::STRING);
-  if (end < 0 || end > strings.size()) end = strings.size();
+  if (input.is_empty()) { return make_empty_column(type_id::STRING); }
   CUDF_EXPECTS(((start >= 0) && (start < end)), "Invalid start parameter value.");
   auto const strings_count  = end - start;
-  auto const offsets_offset = start + strings.offset();
+  auto const offsets_offset = start + input.offset();
 
   // slice the offsets child column
   auto offsets_column = std::make_unique<cudf::column>(
     cudf::detail::slice(
-      strings.offsets(), {offsets_offset, offsets_offset + strings_count + 1}, stream)
+      input.offsets(), {offsets_offset, offsets_offset + strings_count + 1}, stream)
       .front(),
     stream,
     mr);
   auto const chars_offset =
-    offsets_offset == 0 ? 0 : cudf::detail::get_value<int32_t>(offsets_column->view(), 0, stream);
+    offsets_offset == 0 ? 0L : get_offset_value(offsets_column->view(), 0, stream);
   if (chars_offset > 0) {
     // adjust the individual offset values only if needed
-    auto d_offsets = offsets_column->mutable_view();
+    auto d_offsets =
+      cudf::detail::offsetalator_factory::make_output_iterator(offsets_column->mutable_view());
+    auto input_offsets =
+      cudf::detail::offsetalator_factory::make_input_iterator(input.offsets(), offsets_offset);
     thrust::transform(rmm::exec_policy(stream),
-                      d_offsets.begin<int32_t>(),
-                      d_offsets.end<int32_t>(),
-                      d_offsets.begin<int32_t>(),
-                      cuda::proclaim_return_type<int32_t>(
+                      input_offsets,
+                      input_offsets + offsets_column->size(),
+                      d_offsets,
+                      cuda::proclaim_return_type<int64_t>(
```

Comment:
Do you think we should declare something like `size_type` that would be a proper type name for all the `int64_t`s that are being sprinkled around? It's pretty subtle in some cases, like the `0L` literal written above. I would want to reduce the possibility of unwanted casting (like forgetting to write `0L` instead of `0`) by having a type name with clear intent.

Reply:
I do not wish to create a special type at this time and prefer not to obfuscate `int64_t` in this code.